contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinWOOFi.sol

```diff
@@ -0,0 +1,132 @@
+
+}
+
+contract MixinWOOFi{
+
+    using LibERC20TokenV06 for IERC20TokenV06;
+    using LibERC20TokenV06 for IEtherTokenV06;
+    using LibSafeMathV06 for uint256;
+
+    uint internal constant UINT_MAX = type(uint).max;
+
+    // /// @dev Swaps an exact amount of input tokens for as many output tokens as possible.
+    // /// @param _amountIn Amount of input tokens to send
+    // /// @param _minAmountOut The minimum amount of output tokens that must be received for the transaction not to revert.
+    // /// @param _tokenIn Input token
+    // /// @param _tokenOut Output token
+    // /// @param _to recipient of tokens
+    // /// @param pool WOOFi pool where the swap will happen
+    function _swap(
+        uint _amountIn, 
+        uint _minAmountOut, 
+        IERC20TokenV06 _tokenIn, 
+        IERC20TokenV06 _tokenOut, 
+        address _to,
+        address pool
+    ) internal {
+        uint realToAmount;
+        IERC20TokenV06 quoteToken = IWooPP(pool).quoteToken();
+        if (_tokenIn == quoteToken) {
+            realToAmount = IWooPP(pool).sellQuote(
+                _tokenOut,
+                _amountIn,
+                _minAmountOut,
+                _to,
+                0xBfdcBB4C05843163F491C24f9c0019c510786304
+            );
+        } else if (_tokenOut == quoteToken) {
+            realToAmount = IWooPP(pool).sellBase(
+                _tokenIn, 
+                _amountIn, 
+                _minAmountOut, 
+                _to, 
+                0xBfdcBB4C05843163F491C24f9c0019c510786304
+            );
+        } else {
+            uint256 quoteAmount = IWooPP(pool).sellBase(
+                _tokenIn, 
+                _amountIn, 
+                0, 
+                address(this), 
+                0xBfdcBB4C05843163F491C24f9c0019c510786304
+            );
+            realToAmount = IWooPP(pool).sellQuote(
+                _tokenOut, 
+                quoteAmount, 
+                _minAmountOut, 
+                _to, 
+                0xBfdcBB4C05843163F491C24f9c0019c510786304
+            );
+        }
+    }
+
+    function _tradeWOOFi(
+        IERC20TokenV06 sellToken,
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        bytes memory bridgeData
+    )
+        public
+        returns (uint256 boughtAmount)
+    {
+        (address _pool) = abi.decode(bridgeData, (address));
+
+        // Grant the  pool an allowance to sell the first token.
+        sellToken.approveIfBelow(address(_pool), sellAmount);
+        //track the balance to know how much we bought
+        uint256 beforeBalance = buyToken.balanceOf(address(this));
+        _swap(
+            sellAmount,
+            1,
+            sellToken,
+            buyToken,
+            address(this),
+            _pool
+        );
+
+        //calculate the difference in balance from preswap->postswap to find how many tokens out
+        boughtAmount = buyToken.balanceOf(address(this)).safeSub(beforeBalance);
```

Comment:
Looks like `sellBase` and `sellQuote` returns the `boughtAmount`.
Is there a reason to not use that directly? (vs calculating from before and after balances).